sentry/src/main/java/io/sentry/Scope.java

```diff
@@ -83,21 +84,51 @@ public void setLevel(@Nullable SentryLevel level) {
   }
 
   /**
-   * Returns the Scope's transaction
+   * Returns the Scope's transaction name.
    *
    * @return the transaction
    */
-  public @Nullable String getTransaction() {
+  public @Nullable String getTransactionName() {
+    final SentryTransaction tx = this.transaction;
+    return tx != null ? tx.getTransaction() : null;
+  }
+
+  /**
+   * Sets the Scope's transaction.
+   *
+   * @param transaction the transaction
+   */
+  public void setTransaction(final @NotNull String transaction) {
+    final SentryTransaction tx = this.transaction;
+    if (tx != null) {
+      tx.setName(transaction);
+    }
+  }
+
+  /**
+   * Returns current active Span or Transaction.
+   *
+   * @return current active Span or Transaction or null if transaction has not been set.
+   */
+  @Nullable
+  ISpan getSpan() {
+    if (transaction != null && !transaction.getSpans().isEmpty()) {
+      final Span span = transaction.getLatestActiveSpan();
+
+      if (span != null) {
+        return span;
+      }
```

Comment:
All of this API surface will be accessed concurrently on Android (single Hub)
So as I read set setTransaction, clearTransaction etc it will not work.

here for example on line 116 the access to the field `transaction` could throw NPE because `clearTransaction` was called on another thread.

Might be the case that we just make these methods thread-safe.

So things like `startTransaction` will atomically close a previous one if one existed etc.
Won't block this PR though, just a note.

Reply:
I refactored it in a way that clearing transaction won't make this method fail.